# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1315354319 14400
#      Tue Sep 06 20:11:59 2011 -0400
# Branch stable
# Node ID 8e61aa0fac77ac84b198ac41c74141ef4fd50199
# Parent  c7512d0d52e8cb862819e09415e7f6e990f6cab0
src/graphics.cc: Fix mistake in changeset c7512d0d52e8.

diff --git a/src/graphics.cc b/src/graphics.cc
--- a/src/graphics.cc
+++ b/src/graphics.cc
@@ -5122,17 +5122,18 @@ axes::properties::calc_ticks_and_lims (a
     }
 
   double tick_sep = calc_tick_sep (lo , hi);
 
   if (is_logscale && ! (xisinf (hi) || xisinf (lo)))
     {
       // FIXME - what if (hi-lo) < tick_sep?
       //         ex: loglog ([1 1.1])
-      tick_sep = std::ceil (tick_sep, 1.);
+      tick_sep = std::max (tick_sep, 1.);
+      tick_sep = std::ceil (tick_sep);
     }
 
   int i1 = static_cast<int> (gnulib::floor (lo / tick_sep));
   int i2 = static_cast<int> (std::ceil (hi / tick_sep));
 
   if (limmode_is_auto)
     {
       // adjust limits to include min and max tics
